soc/xtensa/intel_adsp/common/include/intel_adsp_ipc.h

```diff
@@ -174,4 +174,18 @@ bool intel_adsp_ipc_send_message_sync(const struct device *dev,
 void intel_adsp_ipc_send_message_emergency(const struct device *dev, uint32_t data,
 					   uint32_t ext_data);
 
+/** @brief Check for pending ACK.
+ *
+ * Returns true if driver expect acknowledgment from host after sending ipc message to it.
+ *
+ * @param dev IPC device.
+ * @return True if driver expect acknowledgment from HOST.
+ */
+static inline bool intel_adsp_ipc_is_ack_pending(const struct device *dev)
+{
+	struct intel_adsp_ipc_data *devdata = dev->data;
+
+	return devdata->tx_ack_pending;
```

Comment:
This looks racy.  What does the caller do with this information?  The only safe thing to do is poll for completion, which the driver already supports, no?

Reply:
Yes, it is intended for poll only and yes, driver checks this before sending new message. Thing I want to do here is to resolve a problem that we have in SOF. Because after message was send and if there is no other messages we are going to Idle thread. If PG is enabled, we will enter D0i3 state and it can happen before host will give ACK after receiving last message.